com.playeveryware.eos/Runtime/Core/Utility/FileSystemUtility.cs

```diff
@@ -455,20 +465,31 @@ public static void ConvertDosToUnixLineEndings(string srcFilename, string destFi
 
         #endregion
 #endif
+
+        #region File Read Functionality
+
+        public static async Task<(bool Success, string Result)> TryReadAllTextAsync(string filePath)
```

Comment:
Hhhuh. Is this a Task that returns an anonymous function? I hadn't considered that as something possible. It's like a named Tuple... I feel like I'd like this more as a struct with these fields.

Reply:
It's kind of a cool C#-y thing to do. 

It returns a `Tuple`, not an anonymous function. 

The equivalent function signature _without_ naming the tuple elements would be:

```csharp
public static async Task<(bool, string)> TryReadAllTextAsync(string filePath)
```

Naming the components helps avoid less than clear code when handling the return value which (without the tuple naming) would require the following syntax when utilizing the returned (and unnamed) `Tuple`:

```csharp
var result = await FileSystemUtility.TryReadAllTextAsync('SomeFile.txt');
bool didReadSucceed = result.item1;
string fileContents = result.item2;

if (didReadSucceed)
{
    /* ... */
}
```